features/bootstrap/SwaggerContext.php

```diff
@@ -0,0 +1,260 @@
+     */
+    public function assertNbPropertiesExist($nb, $className)
+    {
+        $properties = $this->getProperties($className);
+
+        \PHPUnit_Framework_Assert::assertEquals($nb, count($properties));
+    }
+
+    /**
+     * @Then ":prop" property doesn't exist for the Swagger class ":class"
+     */
+    public function assertPropertyNotExist($propertyName, $className)
+    {
+        try {
+            $this->getProperty($propertyName, $className);
+
+            throw new \PHPUnit_Framework_ExpectationFailedException(
+                sprintf('The property "%s" for the class "%s" exist.', $propertyName, $className)
+            );
+        } catch (\Exception $exception) {
+            // an exception must be catched
+        }
+    }
+
+    /**
+     * @Then ":prop" property is required for Swagger class ":class"
+     */
+    public function assertPropertyIsRequired(string $propertyName, string $className)
+    {
+        $classInfo = $this->getClassInfos($className);
+        if (!in_array($propertyName, $classInfo->{'required'})) {
+            throw new \Exception(sprintf('Property "%s" of class "%s" is not required', $propertyName, $className));
+        }
+    }
+
+    /**
+     * @param string $propertyName
+     * @param string $className
+     *
+     * @throws Exception
+     *
+     * @return array | stdClass
+     */
+    private function getProperty(string $propertyName, string $className) : stdClass
+    {
+        $properties = $this->getProperties($className);
+        $propertyInfos = null;
+        foreach ($properties as $classPropertyName => $property) {
+            if ($classPropertyName === $propertyName) {
+                $propertyInfos = $property;
+            }
+        }
+
+        if (empty($propertyInfos)) {
+            throw new \Exception(sprintf('Property "%s" of class "%s" does\'nt exist', $propertyName, $className));
+        }
+
+        return $propertyInfos;
+    }
+
+    /**
+     * Gets an operation by its method name.
+     *
+     * @param string $className
+     * @param string $method
+     *
+     * @throws Exception
+     *
+     * @return array | stdClass
+     */
+    private function getOperation(string $method, string $className) : stdClass
+    {
+        foreach ($this->getOperations($className) as $classMethod => $operation) {
+            if ($classMethod === $method) {
+                return $operation;
+            }
+        }
+
+        throw new \Exception(sprintf('Operation "%s" of class "%s" doesn\'t exist.', $method, $className));
+    }
+
+    /**
+     * Gets all operations of a given class.
+     *
+     * @param string $className
+     *
+     * @throws Exception
+     *
+     * @return array | stdClass
+     */
+    private function getOperations(string $className) : stdClass
+    {
+        $classInfos = $this->getClassInfos($className);
+
+        return isset($classInfos) ? $classInfos : [];
```

Comment:
don't we have something better than `isset` here? `$classInfo` is an array right? so we can use `empty` instead